src/lib/swapper/swappers/OsmosisSwapper/endpoints.ts

```diff
@@ -0,0 +1,210 @@
+import type { Result } from '@sniptt/monads'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { v4 as uuid } from 'uuid'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import type {
+  GetTradeQuoteInput,
+  GetUnsignedTxArgs,
+  SwapErrorRight,
+  Swapper2Api,
+  TradeQuote2,
+} from 'lib/swapper/api'
+import { SwapError, SwapErrorType } from 'lib/swapper/api'
+import type { SymbolDenomMapping } from 'lib/swapper/swappers/OsmosisSwapper/utils/helpers'
+import {
+  buildApiTradeTx,
+  buildPerformIbcTransferUnsignedTx,
+  symbolDenomMapping,
+} from 'lib/swapper/swappers/OsmosisSwapper/utils/helpers'
+import { createDefaultStatusResponse } from 'lib/utils/evm'
+
+import { getTradeQuote } from './getTradeQuote/getTradeQuote'
+import { atomOnOsmosisAssetId, COSMO_OSMO_CHANNEL } from './utils/constants'
+import type { OsmosisSupportedChainId } from './utils/types'
+
+const tradeQuoteMetadata: Map<string, { chainId: OsmosisSupportedChainId }> = new Map()
+
+export const osmosisApi: Swapper2Api = {
+  getTradeQuote: async (
+    input: GetTradeQuoteInput,
+    { sellAssetUsdRate }: { sellAssetUsdRate: string },
+  ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
+    const tradeQuoteResult = await getTradeQuote(input, { sellAssetUsdRate })
+
+    return tradeQuoteResult.map(tradeQuote => {
+      const { receiveAddress, affiliateBps } = input
+      const id = uuid()
+      // TODO(gomes): getTradeQuote() is currently a hack, which represents the swap as a single trade, and needs to be revamped
+      // This effectively means that getUnsignedTx() won't be able to be implemented for all steps, but only the first for now
+      // i.e either the IBC transfer or swap-exact-amount-in
+      tradeQuoteMetadata.set(id, {
+        chainId: tradeQuote.steps[0].sellAsset.chainId as OsmosisSupportedChainId,
+      })
+      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+    })
+  },
+
+  getUnsignedTx: async ({
+    from,
+    tradeQuote,
+    stepIndex,
+  }: GetUnsignedTxArgs): Promise<CosmosSignTx> => {
+    const { accountNumber, buyAsset, sellAsset, sellAmountBeforeFeesCryptoBaseUnit } =
+      tradeQuote.steps[stepIndex]
+    const { receiveAddress } = tradeQuote
+
+    const sellAssetIsOnOsmosisNetwork = sellAsset.chainId === osmosisChainId
+
+    const sellAssetDenom = symbolDenomMapping[sellAsset.symbol as keyof SymbolDenomMapping]
+    const buyAssetDenom = symbolDenomMapping[buyAsset.symbol as keyof SymbolDenomMapping]
+
+    const adapterManager = getChainAdapterManager()
+    const osmosisAdapter = adapterManager.get(osmosisChainId) as osmosis.ChainAdapter | undefined
+    const cosmosAdapter = adapterManager.get(cosmosChainId) as cosmos.ChainAdapter | undefined
+
+    const { REACT_APP_OSMOSIS_NODE_URL: osmoUrl } = getConfig()
+
+    if (!cosmosAdapter || !osmosisAdapter) throw new Error('Failed to get adapters')
```

Comment:
might be handy to create an `assertGetCosmosSdkChainAdapter` helper so these checks arent in multiple places

Reply:
Will add that in the refactor follow-up since this is effectively only boilerplate that will go 🙏🏽

Reply:
https://github.com/shapeshift/web/pull/4905/commits/62703de9e849401f457894cdb3db00f9f30a6a11